src/AutoRest.CSharp/Common/Input/Source/CompilationInput.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using AutoRest.CSharp.Generation.Types;
+using Microsoft.CodeAnalysis;
+
+namespace AutoRest.CSharp.Input.Source
+{
+    public abstract class CompilationInput
+    {
+        protected Compilation _compilation;
+
+        public CompilationInput(Compilation compilation)
+        {
+            _compilation= compilation;
+        }
+
+        private protected abstract void FilterSymbols();
```

Comment:
we could actually remove this method and let each derived class to handle this by themselves